src/test/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSRIntegrationTest.java

```diff
@@ -90,18 +96,82 @@ public void testApplyBQSR(ABQSRTest params) throws IOException {
         }
-        if (params.args != null) {
-            Stream.of(params.args).forEach(arg -> args.add(arg));
+    }
+
+    @Test(dataProvider = "ApplyBQSRTest")
+    public void testApplyBQSRPath(ABQSRTest params) throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path outPath = jimfs.getPath("applyBQSRTest"+params.outputExtension);
+
+            final ArrayList<String> args = new ArrayList<>();
+            Path refPath = null;
+
+            args.add("-I");
+            args.add(new File(params.bam).getAbsolutePath());
+            args.add("--" + StandardArgumentDefinitions.BQSR_TABLE_LONG_NAME);
+            args.add(new File(resourceDir + "HiSeq.20mb.1RG.table.gz").getAbsolutePath());
+            args.add("-O"); args.add(outPath.toUri().toString());
+            if (params.reference != null) {
+                File refFile = new File(params.reference);
+                args.add("-R"); args.add(refFile.getAbsolutePath());
+                refPath = refFile.toPath();
+            }
+            if (params.args != null) {
+                Stream.of(params.args).forEach(arg -> args.add(arg));
+            }
+
+            runCommandLine(args);
+
+            SamAssertionUtils.assertSamsEqual(outPath, new File(params.expectedFile).toPath(), refPath);
         }
+    }
 
-        runCommandLine(args);
+    @Test(dataProvider = "ApplyBQSRTest", groups={"cloud", "bucket"})
+    public void testApplyBQSRCloud(ABQSRTest params) throws IOException {
+        final String outString = BucketUtils.getTempFilePath(getGCPTestStaging() + "tmp/testApplyBQSRCloud",  params.outputExtension);
+        final Path outPath = BucketUtils.getPathOnGcs(outString);
+        try {
+            final ArrayList<String> args = new ArrayList<>();
+            Path refPath = null;
 
-        SamAssertionUtils.assertSamsEqual(outFile, new File(params.expectedFile), refFile);
+            args.add("-I");
+            args.add(new File(params.bam).getAbsolutePath());
+            args.add("--" + StandardArgumentDefinitions.BQSR_TABLE_LONG_NAME);
+            args.add(new File(resourceDir + "HiSeq.20mb.1RG.table.gz").getAbsolutePath());
+            args.add("-O");
+            args.add(outString);
+            if (params.reference != null) {
+                File refFile = new File(params.reference);
+                args.add("-R");
+                args.add(refFile.getAbsolutePath());
+                refPath = refFile.toPath();
+            }
+            if (params.args != null) {
+                Stream.of(params.args).forEach(arg -> args.add(arg));
+            }
+
+            runCommandLine(args);
+
+            SamAssertionUtils.assertSamsEqual(outPath, new File(params.expectedFile).toPath(), refPath);
+        } finally {
+            if (Files.exists(outPath)) {
+                Files.delete(outPath);
```

Comment:
Doesn't `BucketUtils.getTempFilePath()` already mark this for deletion on JVM exit?

Reply:
Oh yes, you're right! I removed the delete and the try/finally since they are redundant. I also added a comment to help in case others also forget.